src/gui/qgsnewdatabasetablenamewidget.cpp

```diff
@@ -0,0 +1,266 @@
+/***************************************************************************
+  qgsnewdatabasetablenamewidget.cpp - QgsNewDatabaseTableNameWidget
+
+ ---------------------
+ begin                : 9.3.2020
+ copyright            : (C) 2020 by Alessandro Pasotti
+ email                : elpaso at itopen dot it
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include <QTreeWidgetItemIterator>
+
+#include "qgsnewdatabasetablenamewidget.h"
+#include "qgsapplication.h"
+#include "qgsdataitemproviderregistry.h"
+#include "qgsdataitemprovider.h"
+#include "qgsproviderregistry.h"
+#include "qgsprovidermetadata.h"
+
+QgsNewDatabaseTableNameWidget::QgsNewDatabaseTableNameWidget(
+  QgsBrowserGuiModel *browserModel,
+  const QStringList &providersFilter,
+  QWidget *parent )
+  : QWidget( parent )
+{
+
+  // Initalize the browser
+  if ( ! browserModel )
+  {
+    mBrowserModel = new QgsBrowserGuiModel( this );
+    mBrowserModel->initialize();
+  }
+  else
+  {
+    mBrowserModel = browserModel;
+    mBrowserModel->initialize();
+  }
+
+  setupUi( this );
+
+  mValidationResults->setStyleSheet( QStringLiteral( "* { font-weight: bold; color: red; }" ) );
```

Comment:
@nirvn is this the safe?

Reply:
It's 'safe' in that we do this in many (relatively speaking) other places.

In the future, we should have QgsApplication::color( Role ), with default colors that can be overwritten by themes. But that won't happen today :)

@elpaso , BTW, I'd request two UI changes here: 
1/ don't use bold, a change of color to stay table name collision is enough;
2/ IMHO, using red to provide a tip to the user as to what to do here (i.e. 'select a database schema, enter a unique name...') is not good UI/UX. Could you change this so that unless you're reporting an error (i.e. name collision), the string uses the default system color? Thanks